File: android/src/main/kotlin/com/siiprinter/sii_printer_plugin/DeviceListItem.java
package com.siiprinter.sii_printer_plugin;


/**
 * Implement items in the device list.
 */
public class DeviceListItem {

    /** device name */
    private final String mDeviceName;

    /** device mac address */
    private final String mDeviceMacAddress;

    /** device ip address */
    private final String mDeviceIpAddress;


    /**
     * constructor
     *
     * @param deviceName
     * @param deviceAddress
     */
    public DeviceListItem(
            final String deviceName,
            final String deviceAddress
    ) {

        mDeviceName = deviceName;
        mDeviceMacAddress = deviceAddress;
        mDeviceIpAddress = "";
    }


    /**
     * constructor
     *
     * @param deviceName
     * @param macAddress
     * @param ipAddress
     */
    public DeviceListItem(
            final String deviceName,
            final String macAddress,
            final String ipAddress
    ) {

        mDeviceName = deviceName;
        mDeviceMacAddress = macAddress;
        mDeviceIpAddress = ipAddress;
    }


    public String getName() {
        return mDeviceName;
    }


    public String getMacAddress() {
        return mDeviceMacAddress;
    }


    public String getIpAddress() {
        return mDeviceIpAddress;
    }

}

File: android/src/main/kotlin/com/siiprinter/sii_printer_plugin/DeviceListItem.kt
package com.siiprinter.sii_printer_plugin

/**
 * Implement items in the device list.
 */
class DeviceListItem {
    /** device name  */
    val name: String

    /** device mac address  */
    val macAddress: String

    /** device ip address  */
    val ipAddress: String

    /**
     * constructor
     *
     * @param deviceName
     * @param deviceAddress
     */
    constructor(
        deviceName: String,
        deviceAddress: String
    ) {
        name = deviceName
        macAddress = deviceAddress
        ipAddress = ""
    }

    /**
     * constructor
     *
     * @param deviceName
     * @param macAddress
     * @param ipAddress
     */
    constructor(
        deviceName: String,
        macAddress: String,
        ipAddress: String
    ) {
        name = deviceName
        this.macAddress = macAddress
        this.ipAddress = ipAddress
    }

    fun toMap(): HashMap<String, Any> {
        return hashMapOf(
            "device_name" to name,
            "mac_address" to macAddress,
            "ip_address" to ipAddress
        )
    }
}